test/test_walreceiver.py

```diff
@@ -110,3 +110,15 @@ def read_message(cls):
         switch_wal(conn)
         wait_for_xlog(pghoard, 1)
         conn.close()
+
+    def test_walreceiver_multiple_timelines(self, recovery_db, pghoard_walreceiver_recovery):
+        """As we want to fetch all timeline history files when starting up, promote a PG instance
+           to bump the timeline and create a history file.
+        """
+        recovery_db.run_cmd("pg_ctl", "-D", recovery_db.pgdata, "promote")
+        pghoard = pghoard_walreceiver_recovery
+        node = pghoard.config["backup_sites"][pghoard.test_site]["nodes"][0]
+        pghoard.start_walreceiver(pghoard.test_site, node, None)
+        with recovery_db.connect() as conn:
+            switch_wal(conn)
+            wait_for_xlog(pghoard, 1)
```

Comment:
This increases coverage but there are no `assert` here, this looks like the code under test could be entirely removed and the test would still pass.

Reply:
There's an assert in `wait_for_xlog`. But yeah, makes sense to check that the newly created timeline history file is really backed up.